appguardrail_core/metrics.py

```
@@ -24,82 +24,199 @@ class SaleReadinessInputs:
 
-@dataclass(frozen=True)
+    def __init__(
+        self,
+        metric_id: str | None = None,
+        metric_label: str | None = None,
+        metric_value: float | int | bool | None = None,
+        target_threshold: str | None = None,
+        target_passed: bool | None = None,
+        readiness_pillar: str | None = None,
+        *,
+        id: str | None = None,
```

Ruff (0.16.3)

[error] Function argument `id` is shadowing a Python builtin

(A002)